src/main/java/com/matera/cadastrocentral/telephone/TelephoneService.java

```diff
@@ -67,15 +71,29 @@ public void deleteTelephone(UUID telephoneId) {
     public Telephone alterTelephoneByTelephoneId(TelephoneDTO telephone) {
         Optional<Telephone> optionalTelephone = telephoneRepository.findById(telephone.getTelephoneId());
         if (optionalTelephone.isPresent()) {
-            Telephone auxTelephone = optionalTelephone.get();
-            auxTelephone.setClientId(Optional.ofNullable(telephone.getClientId())
-                                    .orElse(optionalTelephone.get().getClientId()));
-            auxTelephone.setTelephoneTypeId(Optional.of(telephone.getTelephoneTypeId())
-                                    .orElse(optionalTelephone.get().getTelephoneTypeId()));
+            return telephoneRepository.save(new Telephone(telephone));
+        } else {
+            throw new TelephoneNotFound();
+        }
+    }
+
+    public Telephone patchTelephoneByTelephoneId(TelephoneDTO telephone) {
+        ClientService clientService = new ClientService(clientRepository, null);
+        if (telephoneRepository.findById(telephone.getTelephoneId()).isPresent()) {
+            Telephone auxTelephone = telephoneRepository.findById(telephone.getTelephoneId()).get();
+            // Check if client input exists
+            if (telephone.getClientId() != null) {
+                if (clientService.getClientById(telephone.getClientId()).isPresent()) {
+                    auxTelephone.setClientId(telephone.getClientId());
+                }
+            }
+            if (telephone.getTelephoneTypeId() != null)
+                auxTelephone.setTelephoneTypeId(Optional.of(telephone.getTelephoneTypeId())
+                    .orElse(auxTelephone.getTelephoneTypeId()));
             auxTelephone.setNumber(Optional.ofNullable(telephone.getNumber())
```

Comment:
Using optional to validate null/empty fields is not a good idea. you can add this lib to project pom https://mvnrepository.com/artifact/org.apache.commons/commons-lang3/3.11

And use it like this

```
// StringUtils.isNotBlank checks for null/empty strings an all spaces strings like "    "
if(StringUtils.isNotBlank(telephone.getNumber())) {
   auxTelephone.setNumber(telephone.getNumber());
}
```